requests_oauth2client/dpop.py

```diff
@@ -139,6 +139,7 @@ def _response_hook(self, response: requests.Response, **kwargs: Any) -> requests
         ):
             self.dpop_key.handle_rs_provided_dpop_nonce(response)
             request = self(response.request.copy())
+            request.deregister_hook("response", self._response_hook)
```

Comment:
Hmm, I'm realizing that this only removes the *first* such hook, and the call to `self(…)` will add a second copy of it, so you'll need to either:
- call this in a loop until it returns false, or
- add an optional arg to `__call__` that disables registering the hook, or
- just call `add_dpop_proof` here instead of calling `self` (I don't think the extra call to `super` should be necessary either?)